### PR TITLE
.github: Created GitHub actions workflow, and deleted travis.

### PR DESCRIPTION
**Pull request by cjyar**
Migrating this repo from Travis to GitHub actions.

### CODE REVIEW

**Comment by clintfred** [comment]
Not necessary, but it's probably also easy to add a security-audit action. See https://github.com/IronCoreLabs/gridiron/blob/master/.github/workflows/security-audit.yaml

**Comment by cjyar** [comment]
@clintfred Good idea.

**Comment by cjyar** [comment]
Please see sample release at https://github.com/IronCoreLabs/ironoxide-java/releases/tag/v0.12.1-test. I manually marked that as a prerelease. We can delete that test release once this PR is accepted.

**Comment by cjyar** [comment]
Comparing old (Travis) and new (GitHub) builds:
```
-rw-r--r--@ 1 chris  staff  11238216 Mar  9 09:42 libironoxide_java.so.new
-rw-r--r--@ 1 chris  staff  11270736 Mar  9 09:43 libironoxide_java.so.old
```

New one is built with `stable`, which for this build was rustc 1.41.1. Travis used 1.41.0.

Verified the lib works by using it in `sbt test` in `ironoxide-scala`.

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        restore-keys: |
+          ${{ github.workflow }}-sbt-
+        path: ~/.sbt
+    - name: Decrypt keys for unit tests
+      uses: IronCoreLabs/ironhide-actions/decrypt@v1
+      if: matrix.target != 'macos-10.15'
+      with:
+        keys: ${{ secrets.IRONHIDE_KEYS }}
+        input: tests/src/test/resources/service-keys.conf.iron
+    - name: Decrypt keys for unit tests
```
Comment (clintfred) [inline .github/workflows/ci.yaml]:
```suggestion
    - name: Decrypt keys for unit tests (MacOS)
```

**Comment by clintfred** [comment]
https://github.com/IronCoreLabs/ironoxide-java/actions/runs/52384459

Looks like there are some LLVM warnings. Not sure if those are important...

**Comment by cjyar** [comment]
@clintfred There are no looping constructs in GHA, other than the matrix build (which GHA calls a "strategy"). Something has to gather the results from all those matrix jobs. Sadly, the `download-artifact` action can only handle one artifact at a time, so we can't use a wildcard and catch them all in a single step. Same story for `upload-release-asset`.

I don't know if the LLVM warnings are important. We can probably get rid of them by installing `llvm` on the builders.

**Comment by cjyar** [comment]
Got rid of the LLVM warnings, except for Mac where it seemed safer to leave the warning than "fix" it.

**Pull request merged**